csharp/ql/test/library-tests/cil/consistency/Handles.ql

```diff
@@ -0,0 +1,51 @@
+import csharp
+import cil
+import dotnet
+
+class MetadataEntity extends DotNet::NamedElement, @metadata_entity {  
+  int getHandle() { metadata_handle(this, _, result) }
+  
+  predicate hasHandle() { exists(getHandle()) }
+  
+  Assembly getAssembly() { metadata_handle(this, result, _) }
+}
+
+query predicate tooManyMatchingHandles(MetadataEntity e) {
+  count(MetadataEntity e2 | e.matchesHandle(e2))>2
+}
+
+query predicate missingCil(Element e) {
```

Comment:
Change type to `MetadataEntity` and remove infix cast below

Reply:
Then I would need an infix cast in a different place - I think on the `fromLibrary`.